pkg/sfu/downtrack.go

```diff
@@ -854,76 +857,100 @@ func (d *DownTrack) CreateSenderReport() *rtcp.SenderReport {
-		frameRate = 50
-	}
+		frameRate := uint32(30)
+		if d.mime == "audio/opus" {
+			frameRate = 50
+		}
 
-	// send a number of blank frames just in case there is loss.
-	// Intentionally ignoring check for mute or bandwidth constrained mute
-	// as this is used to clear client side buffer.
-	numFrames := int(float32(frameRate) * duration)
-	snts, frameEndNeeded, err := d.forwarder.GetSnTsForBlankFrames(frameRate, numFrames)
-	if err != nil {
-		d.logger.Warnw("could not get SN/TS to write blank frames", err)
-		return err
-	}
+		// send a number of blank frames just in case there is loss.
+		// Intentionally ignoring check for mute or bandwidth constrained mute
+		// as this is used to clear client side buffer.
+		numFrames := int(float32(frameRate) * duration)
+		frameDuration := time.Duration(1000/frameRate) * time.Millisecond
 
-	for i := 0; i < len(snts); i++ {
-		hdr := rtp.Header{
-			Version:        2,
-			Padding:        false,
-			Marker:         true,
-			PayloadType:    d.payloadType,
-			SequenceNumber: snts[i].sequenceNumber,
-			Timestamp:      snts[i].timestamp,
-			SSRC:           d.ssrc,
-			CSRC:           []uint32{},
-		}
+		ticker := time.NewTicker(frameDuration)
+		defer ticker.Stop()
 
-		err = d.writeRTPHeaderExtensions(&hdr)
-		if err != nil {
-			d.logger.Warnw("could not write header extensions for blank frame", err)
-			return err
-		}
+		for {
+			if generation != d.blankFramesGeneration.Load() || numFrames <= 0 {
+				close(done)
+				return
+			}
 
-		var pktSize int
-		switch d.mime {
-		case "audio/opus":
-			pktSize, err = d.writeOpusBlankFrame(&hdr, frameEndNeeded)
-		case "video/vp8":
-			pktSize, err = d.writeVP8BlankFrame(&hdr, frameEndNeeded)
-		case "video/h264":
-			pktSize, err = d.writeH264BlankFrame(&hdr, frameEndNeeded)
-		default:
-			return nil
-		}
-		if err != nil {
-			d.logger.Warnw("could not write blank frame", err)
-			return err
-		}
+			snts, frameEndNeeded, err := d.forwarder.GetSnTsForBlankFrames(frameRate, 1)
+			if err != nil {
+				d.logger.Warnw("could not get SN/TS for blank frame", err)
+				close(done)
+				return
+			}
 
-		for _, f := range d.onPacketSent {
-			f(d, pktSize)
-		}
+			for i := 0; i < len(snts); i++ {
+				hdr := rtp.Header{
+					Version:        2,
+					Padding:        false,
+					Marker:         true,
+					PayloadType:    d.payloadType,
+					SequenceNumber: snts[i].sequenceNumber,
+					Timestamp:      snts[i].timestamp,
+					SSRC:           d.ssrc,
+					CSRC:           []uint32{},
+				}
 
-		// only the first frame will need frameEndNeeded to close out the
-		// previous picture, rest are small key frames
-		frameEndNeeded = false
-	}
+				err = d.writeRTPHeaderExtensions(&hdr)
+				if err != nil {
+					d.logger.Warnw("could not write header extension for blank frame", err)
+					close(done)
+					return
+				}
 
-	return nil
+				var pktSize int
+				switch d.mime {
+				case "audio/opus":
+					pktSize, err = d.writeOpusBlankFrame(&hdr, frameEndNeeded)
```

Comment:
can move the switch case at the head of the function and combine with the unsupport mime check

Reply:
Will check that @cnderrauber .

Reply:
Sorry @cnderrauber . After looking at the code, I do not quite understand the suggested change. Are you suggesting the unsupported check code can use a switch/case instead of the `if`s? Or some form of unrolling this loop?